unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs

```diff
@@ -0,0 +1,200 @@
+        public float cycleTime = 24;
+        public float minutesPerSecond = 60;
+
+        private float timeOfTheDay;                            // (Nishant.K) Time will be provided from outside, So remove this variable
+        private Light directionalLight;
+        private SkyboxConfiguration configuration;
+        private Material selectedMat;
+        private bool overrideDefaultSkybox;
+        private string overrideSkyboxID;
+        private bool isPaused;
+        private float timeNormalizationFactor;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            i = this;
+
+            // Enable/Disable or Create new Directional Light Object
+            directionalLight = GameObject.FindObjectsOfType<Light>().Where(s => s.type == LightType.Directional).FirstOrDefault();
+
+            if (directionalLight == null)
+            {
+                GameObject temp = new GameObject("The Sun");
+                // Add the light component
+                directionalLight = temp.AddComponent<Light>();
+                directionalLight.type = LightType.Directional;
+            }
+
+            timeOfTheDay = 0;
+
+            DataStore.i.skyboxConfig.useProceduralSkybox.Set(true);
+
+            UpdateConfig();
+            DataStore.i.skyboxConfig.objectUpdated.OnChange += UpdateConfig;
+        }
+
+        public void UpdateConfig(bool current = true, bool previous = false)
+        {
+            // Apply configuration
+            overrideDefaultSkybox = true;
+            overrideSkyboxID = DataStore.i.skyboxConfig.configToLoad.Get();
+
+            // Apply time
+            minutesPerSecond = DataStore.i.skyboxConfig.minutesPerSecond.Get();
+
+            // if Paused
+            if (DataStore.i.skyboxConfig.pauseTime.Get())
+            {
+                PauseTime();
+            }
+            else
+            {
+                ResumeTime();
+            }
+
+            // Jump to time
+            if (DataStore.i.skyboxConfig.jumpTime)
+            {
+                float jumpToTime = DataStore.i.skyboxConfig.jumpToTime.Get();
+                timeOfTheDay = Mathf.Clamp(jumpToTime, 0, cycleTime);
+                DataStore.i.skyboxConfig.jumpTime = false;
+            }
+
+            if (DataStore.i.skyboxConfig.useProceduralSkybox.Get())
+            {
+                Applyconfig();
+            }
+            else
+            {
+                RenderProfileManifest.i.currentProfile.Apply();
+            }
+
+            // Reset Object Update value without notifying
+            DataStore.i.skyboxConfig.objectUpdated.Set(false, false);
+        }
+
+        void Applyconfig()
+        {
+            SelectSkyboxConfiguration();
+
+            if (!configuration.useDirectionalLight)
+            {
+                directionalLight.gameObject.SetActive(false);
+            }
+            //DCL.DataStore.i.isProceduralSkyboxInUse.Set(true);
```

Comment:
```suggestion
```

Reply:
Done. Making all these changes in a new commit